tools/old/softirqs.py

```diff
@@ -48,58 +50,107 @@
 debug = 0
 
 # define BPF program
-bpf_text = """
-#include <uapi/linux/ptrace.h>
-
-typedef struct irq_key {
-    u64 ip;
-    u64 slot;
-} irq_key_t;
-BPF_HASH(start, u32);
-BPF_HASH(iptr, u32);
-BPF_HISTOGRAM(dist, irq_key_t);
-
-// time IRQ
-int trace_start(struct pt_regs *ctx)
-{
-    u32 pid = bpf_get_current_pid_tgid();
-    u64 ip = PT_REGS_IP(ctx), ts = bpf_ktime_get_ns();
-    start.update(&pid, &ts);
-    iptr.update(&pid, &ip);
-    return 0;
-}
-
-int trace_completion(struct pt_regs *ctx)
-{
-    u64 *tsp, delta, ip, *ipp;
-    u32 pid = bpf_get_current_pid_tgid();
-
-    // fetch timestamp and calculate delta
-    tsp = start.lookup(&pid);
-    ipp = iptr.lookup(&pid);
-    if (tsp == 0 || ipp == 0) {
-        return 0;   // missed start
+bpf_text = ""
+if args.CPUidx:
```

Comment:
Well, technically, I agree, though, I was refraining from making the replacement section a mess. So I had a tradeoff between readability and code duplication in mind.